src/engine/strat_engine/backstore/metadata/static_header.rs

```diff
@@ -309,12 +337,7 @@ impl StaticHeader {
         ) {
             // We read both copies without an IO error.
             (Ok(buf_loc_1), Ok(buf_loc_2)) => match (buf_loc_1, buf_loc_2) {
-                (Ok(loc_1), Ok(loc_2)) => match (loc_1, loc_2) {
-                    (Some(loc_1), Some(loc_2)) => compare_static_headers(f, loc_1, loc_2),
-                    (None, None) => Ok(None),
-                    (Some(loc_1), None) => write_header(f, loc_1, MetadataLocation::Second),
-                    (None, Some(loc_2)) => write_header(f, loc_2, MetadataLocation::First),
-                },
+                (Ok(loc_1), Ok(loc_2)) => ok_ok_static_header_handling(f, loc_1, loc_2),
```

Comment:
Please get rid of the inline comments just above and also the few below. They were put in when the body of the method was about five times as long as it is ow, to serve as some sort of sign posts. But since all the improvements have been made to the method, they are a bit worse than redundant.